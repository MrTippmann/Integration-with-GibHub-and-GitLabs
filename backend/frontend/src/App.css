.header {
  padding: 40px;
  text-align: center;
  background: #1abc9c;
  color: white;
  font-size: 24px;
}

input[type=text] {
  background-color: white;
  background-image: url('searchicon.jpg');
  background-position: 0px 5px;
  background-repeat: no-repeat;
  background-size: 50px;
  width: 50%;
  height: 60px;
  padding: 20px 50px;
  margin: 20px 10px;
  box-sizing: border-box;
}

input[type=text]:focus {
  border: 3px solid #555;
}

.form {
  background-color: grey;
  color: white;
}

#profilePic {
  width: 300px;
  height: 400px;
}

.userDetails {
  background-color: #2f3836;
}

ul {
  background: white;
  padding: 20px;
  margin: 20px;
  font-size: 20px;;
}

ul li {
  background: #1abc9c;
  color: black;
  margin: 5px;
}
