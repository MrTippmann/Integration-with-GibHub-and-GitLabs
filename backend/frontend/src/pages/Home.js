import React from "react";
import "../App.css";
import UsersGitHub from "../components/UsersGitHub";
import UsersGitLab from "../components/UserGitLab";

class Home extends React.Component {
  constructor(props) {
    super(props);
    this.state = { username: "", formSubmit: false };
    this.handleInput = this.handleInput.bind(this);
    this.handleFormSubmit = this.handleFormSubmit.bind(this);
  }

  // Handles user's input by updating the word to what the user has typed
  handleInput(event) {
    this.setState({
      username: event.target.value,
    });
  }

  //  Handles the submission of the form by marking formSubmit as true
  handleFormSubmit(event) {
    event.preventDefault();
    this.setState({ formSubmit: true });
  }

  // Renders the input field and the def and usage of a word if it has both once the form has been submitted
  render() {
    return (
      <>
        <div className="App">
          <div class="header">
            <h1>GitHub and GitLab users</h1>
          </div>
          <div class="form">
            <form onSubmit={this.handleFormSubmit}>
              <label>
                Search for a user:
                <input
                  type="text"
                  placeholder="Search..."
                  onChange={this.handleInput}
                />
              </label>
            </form>
            <p>Hint --- Press enter once you have typed out your word</p>
          </div>
          {this.state.formSubmit && <UsersGitHub username={this.state.username} />}
          {this.state.formSubmit && <UsersGitLab username={this.state.username} />}
        </div>
      </>
    );
  }
}

export default Home;
