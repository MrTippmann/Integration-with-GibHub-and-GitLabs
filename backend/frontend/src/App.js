import React from "react";
import Home from "./pages/Home";


// React Router does not have any opinions about
// how you should parse URL query strings.
//
// If you use simple key=value query strings and
// you do not need to support IE 11, you can use
// the browser's built-in URLSearchParams API.
//
// If your query strings contain array or object
// syntax, you'll probably need to bring your own
// query parsing function.

function App() {
  return (
    <div>
      <Home />
    </div>
  );
}

export default App;
