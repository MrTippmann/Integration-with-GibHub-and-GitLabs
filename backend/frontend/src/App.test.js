import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./App";
import Home from "./Pages/Home";
import renderer from "react-test-renderer";

jest.mock("./");

// Testing if Dictionary component renders correctly
test("renders correctly", () => {
  const tree = renderer
    .create(<Home page="http://localhost:3000/"></Home>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// Testing that first page has header
test("Header loads correctly", () => {
  render(<App />);
  expect(screen.getByText("GitHub and GitLab users")).toBeInTheDocument();
});

fetch(
  `/api/gitlab-users/MrTippmann`
)

// Suppose to give a runtime error
test('id is id:91426696', done => {
  function callback(error, data) {
    if (error) {
      done(error);
      return;
    }
    try {
      expect(data.id).toBe('id:91426696');
      done();
    } catch (error) {
      done(error);
    }
  }

  fetch(callback);
});