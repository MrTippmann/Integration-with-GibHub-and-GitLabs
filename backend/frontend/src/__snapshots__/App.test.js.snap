// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <div
    class="header"
  >
    <h1>
      GitHub and GitLab users
    </h1>
  </div>
  <div
    class="form"
  >
    <form
      onSubmit={[Function]}
    >
      <label>
        Search for a user:
        <input
          onChange={[Function]}
          placeholder="Search..."
          type="text"
        />
      </label>
    </form>
    <p>
      Hint --- Press enter once you have typed out your word
    </p>
  </div>
</div>
`;
